src/openzeppelin/tests/test_account.cairo

```diff
@@ -0,0 +1,330 @@
+use core::traits::Into;
+use array::ArrayTrait;
+use array::SpanTrait;
+use core::result::ResultTrait;
+use option::OptionTrait;
+use serde::Serde;
+use starknet::class_hash::Felt252TryIntoClassHash;
+use starknet::ContractAddress;
+use starknet::contract_address_const;
+use starknet::syscalls::deploy_syscall;
+use starknet::testing;
+use traits::TryInto;
+
+use openzeppelin::account::Account;
+use openzeppelin::account::IAccountDispatcher;
+use openzeppelin::account::IAccountDispatcherTrait;
+use openzeppelin::account::ERC165_ACCOUNT_ID;
+use openzeppelin::account::ERC1271_VALIDATED;
+use openzeppelin::account::TRANSACTION_VERSION;
+use openzeppelin::account::Call;
+use openzeppelin::token::erc20::ERC20;
+use openzeppelin::token::erc20::IERC20Dispatcher;
+use openzeppelin::token::erc20::IERC20DispatcherTrait;
+use openzeppelin::introspection::erc165::IERC165_ID;
+
+#[derive(Drop)]
+struct SignedTransactionData {
+    private_key: felt252,
+    public_key: felt252,
+    transaction_hash: felt252,
+    r: felt252,
+    s: felt252
+}
+
+fn PUBLIC_KEY() -> felt252 {
+    0x333333
+}
+fn ACCOUNT_ADDRESS() -> ContractAddress {
+    contract_address_const::<0x111111>()
+}
+fn SIGNED_TX_DATA() -> SignedTransactionData {
+    SignedTransactionData {
+        private_key: 1234,
+        public_key: 883045738439352841478194533192765345509759306772397516907181243450667673002,
+        transaction_hash: 2717105892474786771566982177444710571376803476229898722748888396642649184538,
+        r: 3068558690657879390136740086327753007413919701043650133111397282816679110801,
+        s: 3355728545224320878895493649495491771252432631648740019139167265522817576501
+    }
+}
+
+fn setup_dispatcher(data: Option<@SignedTransactionData>) -> IAccountDispatcher {
```

Comment:
```suggestion
fn setup_account(data: Option<@SignedTransactionData>) -> IAccountDispatcher {
```
is this more accurate?

Reply:
thinking more about it, can't we remove the account deployment and call the methods directly from the module, keeping simply `setup_tx_info(info)` where info is today's `SignedTransactionData`?

Reply:
Because the `deploy_syscall` registers the deployed address, and the `call_contract_syscall` panics if the target address is not registered as deployed. So we can't test `execute` for example.